### ISSUE TITLE
Offset for Col on Grid in Grid it inherit values from parent

### ISSUE THREAD

**Issue by dipiash** [mantinedev/mantine#5624]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.4.2

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

<img width="1682" alt="image" src="https://github.com/mantinedev/mantine/assets/14073668/b7f76288-d43f-4f66-af20-5889526e1b04">

```tsx
     <Grid gutter={16}>
        <Grid.Col span={3} bg="blue">
          Col 1
        </Grid.Col>
        <Grid.Col
          offset={{ base: 0, xs: 0, sm: 0, md: 1, lg: 1, xl: 1 }}
          span={7}
        >
          <Grid gutter={16}>
            <Grid.Col span={3} bg="red">
              Col 2
            </Grid.Col>
            <Grid.Col span={7} bg="cyan">
              Col 3
            </Grid.Col>
          </Grid>
        </Grid.Col>
        <Grid.Col span={1}>Col 3</Grid.Col>
      </Grid>
```

Offset for Col on Grid in Grid it inherit values from parent

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/hardcore-shadow-wqp6lz?file=%2Fsrc%2FApp.tsx

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by dipiash** [comment]
@rtivital hi! Let me know please, what do you think about Grid.Col behaviour?

**Comment by rtivital** [comment]
All reported issues are examined and fixed in the general queue. Once the order of the issue is reached, the resolution is provided.

**Comment by dipiash** [comment]
Thank you!

**Comment by rtivital** [comment]
The issue has been fixed in 7.5.0 – https://codesandbox.io/p/sandbox/dawn-darkness-5c2rwy?file=%2Fsrc%2FApp.tsx

**Comment by dipiash** [comment]
@rtivital Sounds great! Thank you for fix 👍
